huntsman/tests/test_camera.py

```diff
@@ -0,0 +1,347 @@
+    except NotImplementedError:
+        pytest.skip("Camera {} doesn't implement temperature control".format(camera.name))
+    else:
+        assert abs(camera.ccd_set_point - 10 * u.Celsius) < 0.5 * u.Celsius
+
+
+def test_enable_cooling(camera):
+    try:
+        camera.ccd_cooling_enabled = True
+    except NotImplementedError:
+        pytest.skip("Camera {} doesn't implement control of cooling status".format(camera.name))
+    else:
+        assert camera.ccd_cooling_enabled is True
+
+
+def test_get_cooling_power(camera):
+    try:
+        power = camera.ccd_cooling_power
+    except NotImplementedError:
+        pytest.skip("Camera {} doesn't implement cooling power readout".format(camera.name))
+    else:
+        assert power is not None
+
+
+def test_disable_cooling(camera):
+    try:
+        camera.ccd_cooling_enabled = False
+    except NotImplementedError:
+        pytest.skip("Camera {} doesn't implement control of cooling status".format(camera.name))
+    else:
+        assert camera.ccd_cooling_enabled is False
+
+
+def test_exposure(camera, tmpdir):
+    """
+    Tests basic take_exposure functionality
+    """
+    fits_path = str(tmpdir.join('test_exposure.fits'))
+    # A one second normal exposure.
+    camera.take_exposure(filename=fits_path)
+    # By default take_exposure is non-blocking, need to give it some time to complete.
+    time.sleep(5)
+    assert os.path.exists(fits_path)
+    # If can retrieve some header data there's a good chance it's a valid FITS file
+    header = fits_utils.getheader(fits_path)
+    assert header['EXPTIME'] == 1.0
+    assert header['IMAGETYP'] == 'Light Frame'
+
+
+def test_exposure_blocking(camera, tmpdir):
+    """
+    Tests blocking take_exposure functionality. At least for now only SBIG cameras do this.
+    """
+    fits_path = str(tmpdir.join('test_exposure_blocking.fits'))
+    # A one second exposure, command should block until complete so FITS
+    # should exist immediately afterwards
+    camera.take_exposure(filename=fits_path, blocking=True)
+    assert os.path.exists(fits_path)
+    # If can retrieve some header data there's a good chance it's a valid FITS file
+    header = fits_utils.getheader(fits_path)
+    assert header['EXPTIME'] == 1.0
+    assert header['IMAGETYP'] == 'Light Frame'
+
+
+def test_exposure_dark(camera, tmpdir):
+    """
+    Tests taking a dark. At least for now only SBIG cameras do this.
+    """
+    fits_path = str(tmpdir.join('test_exposure_dark.fits'))
+    # A 1 second dark exposure
+    camera.take_exposure(filename=fits_path, dark=True, blocking=True)
+    assert os.path.exists(fits_path)
+    # If can retrieve some header data there's a good chance it's a valid FITS file
+    header = fits_utils.getheader(fits_path)
+    assert header['EXPTIME'] == 1.0
+    assert header['IMAGETYP'] == 'Dark Frame'
+
+
+@pytest.mark.filterwarnings('ignore:Attempt to start exposure')
+def test_exposure_collision(camera, tmpdir):
```

Comment:
Still needed here?

Reply:
Hmmm, These Pyro cameras don't emit the warning that the other Camera classes do (shame!), so I don't need the warnings filter. The test is still useful though, just to make sure that trying to start another exposure while one is in progress doesn't break anything.

I just leave it up to Pyro to handle any 'exposure collisions'. I've configured the camera servers to create only a single camera object, to handle remote method calls sequentially ('multiplex' mode), and made the server side `take_exposure` method blocking. If you try to start another exposure while one is in progress Pyro will just put it in a queue until the remote camera is ready.

Reply:
Got it. i was thinking this was testing the collision of multiple cameras (a la the SBIG simultaneous readout issue)